spec/app/tournament/tournament.spec.js

```diff
@@ -0,0 +1,121 @@
+import {Tournament} from "../../../app/tournament/tournament";
+import {Player} from "../../../app/tournament/player";
+
+describe('class Tournament', function() {
+
+    var spiderman = new Player("Spiderman", 1);
+    var batman = new Player("Batman", 3);
+    var theThing = new Player("The Thing", 5);
+
+    var tournament;
+
+    beforeEach( function() {
+        tournament = new Tournament();
+    });
+
+    function setMockedTournament(players) {
+        while (players--) {
+            tournament.addPlayer(spiderman);
+        }
+    }
+
+    it('should return false when searching for a non-enrolled player', function() {
+        expect( tournament.existsPlayer(spiderman) ).toBeUndefined();
+    });
+
+    it('should return a validation error when validating a player who is already enrolled', function() {
+        tournament.addPlayer(spiderman);
+        var validationResult = tournament.validatePlayer(spiderman);
+
+        expect( validationResult.error ).toEqual("Player already enrolled");
+    });
+
+    it('should return an unsuccessful validation when validating a player who is already enrolled', function() {
+        tournament.addPlayer(spiderman);
+        var validationResult = tournament.validatePlayer(spiderman);
+
+        expect( validationResult.success ).toBe(false);
+    });
+
+    it('should not return any validation error when validating a player who is not yet enrolled', function() {
+        tournament.addPlayer(spiderman);
+        var validationResult = tournament.validatePlayer(batman);
+
+        expect( validationResult.error ).toEqual("");
+    });
+
+    it('should return a successful validation when validating a player who is not yet enrolled', function() {
+        tournament.addPlayer(spiderman);
+        var validationResult = tournament.validatePlayer(batman);
+
+        expect( validationResult.success ).toBe(true);
+    });
+
+    it('should create 1 match generating with 2 players', function() {
+        tournament.addPlayer(spiderman);
+        tournament.addPlayer(batman);
+        tournament.generate();
+
+        expect( tournament.matches.length ).toEqual(1);
+    });
+
+    it('should create 3 match generating with 4 players', function() {
+        setMockedTournament(4)
+        tournament.generate();
+
+        expect( tournament.matches.length ).toEqual(3);
+    });
+
+    it('should order its players by rank after adding a new player', function() {
+        tournament.addPlayer(theThing);
+
+        expect( tournament.players[0].rank ).toEqual(5);
+
+        tournament.addPlayer(batman);
```

Comment:
True, but the reason is precisely because those players are not to be modified, since they represent each one a rank (highest, middle and lowest). Otherwise I would have to instantiate spiderman up to 8 times and batman 4.